File: Greedy Algorithms/Easy/Assign_Cookies.java

import java.util.Arrays;

public class Assign_Cookies{
    public static void main(String[] args){
        int[] greed = {1, 5, 3, 3, 4};
        int[] cookie_size= {4, 2, 1, 2, 1, 3};

        int n = greed.length, m = cookie_size.length;
        int l = 0, r = 0;

        Arrays.sort(greed);
        Arrays.sort(cookie_size);

        while (l < m && r < n) { 
            if (greed[r] <= cookie_size[l]) {
                r++;
            }
            l = l + 1;
        }
        System.out.println(r + " children can be satisfied !");
    }
}

File: Two Pointers/Assign_cookies.py
def assign_cookies(children, cookies):
    children.sort()
    cookies.sort()

    child = 0
    cook = 0

    m = len(children)
    n = len(cookies)

    content = 0

    while child < m and cook < n:
        if children[child] <= cookies[cook]:
            content += 1
            child += 1
            cook += 1
        else:
            cook += 1

    return content



children = [1, 2, 3]
cookies = [1, 1]
print(f"{assign_cookies(children, cookies)} can be satisfied")